spring-cloud-gcp-logging/src/test/java/com/google/cloud/spring/logging/StackdriverJsonLayoutLoggerTests.java

```diff
@@ -246,6 +246,22 @@ void testJsonSeverityLevelMapping() {
         .containsExactly("DEBUG", "DEBUG", "INFO", "WARNING", "ERROR", "ERROR");
   }
 
+  @Test
+  void testTimestampNanos() {
+    LOGGER.warn("test1");
+    LOGGER.warn("test2");
+
+    List<String> jsonLogRecords = Arrays.asList(new String(logOutput.toByteArray()).split("\n"));
+
+    List<Double> logTimestampNanos =
+            jsonLogRecords.stream()
+                    .map(record -> GSON.fromJson(record, Map.class))
+                    .map(data -> (Double) data.get(StackdriverTraceConstants.TIMESTAMP_NANOS_ATTRIBUTE))
+                    .collect(Collectors.toList());
+
+    assertThat(logTimestampNanos).anySatisfy(nanos -> assertThat(nanos % 1000000).isGreaterThan(0));
```

Comment:
Can't nanos be `0` though?

Reply:
Yes, I wanted to create test to ensure that there is nano precision and not just mili.
I am testing 2 logs with any greater than 0 to lower false negative possibility (when all micros and nanos are 0) to 1:1e+12.

Not sure if you like it this way or maybe just simple 1 log with nanos greaterOrEqual(0).

Reply:
I think the way you have it is fine. As you point out, the likelihood of flakiness due to this is tiny. You might just want to make sure you are getting 2 timestamps in the array. You can even make the number of logs a variable.